### DISCUSSION TITLE
Can we exclude some modules?

### DISCUSSION THREAD

**Discussion by xiaoqian0408** [cqfn/diKTat#1360] [Q&A] [answered]
Now we use gradle, and diktat plugin, and run diktat on all our modules. Can we exclude some specific modules?
I just want to run diktat on only a few modules.

Thanks

**Comment by petertrr** [comment, marked as answer]
It really depends on how you apply the plugin. By default, when you apply diktat-gradle-plugin, it only checks Kotlin sources in the project where you apply it. Maybe, you've misconfigured inputs of diktat task so that it captures other sources too. It's really hard to say without seeing your configuration.

Following current Gradle best practices, I would suggest to create a [precompiled script plugin](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:precompiled_plugins) with common diktat configuration and apply it only on those gradle modules where you would like it to be.
